pkg/controller/jobframework/validation.go

```diff
@@ -162,28 +163,25 @@ func validateUpdateForMaxExecTime(oldJob, newJob GenericJob) field.ErrorList {
 }
 
 // ValidateImmutablePodSpec function is used for serving workloads to ensure no changes are allowed
-// to the PodSpec except for the image field in containers.
-func ValidateImmutablePodSpec(newPodSpec *corev1.PodSpec, oldPodSpec *corev1.PodSpec, fieldPath *field.Path) field.ErrorList {
-	// handle updateable fields by munging those fields prior to deep equal comparison.
-	mungedPodSpec := newPodSpec.DeepCopy()
-
-	// munge spec.containers[*].image
-	newContainers := make([]corev1.Container, 0, len(newPodSpec.Containers))
-	for i, container := range mungedPodSpec.Containers {
-		container.Image = oldPodSpec.Containers[i].Image
-		newContainers = append(newContainers, container)
-	}
-	mungedPodSpec.Containers = newContainers
+// to the PodSpec except fields that required for role-hash generation.
+func ValidateImmutablePodSpec(newPodSpec corev1.PodSpec, oldPodSpec corev1.PodSpec, fieldPath *field.Path) field.ErrorList {
```

Comment:
ValidateImmutablePodGroupPodSpec this function is specific to pod groups only

Reply:
Done